text/0002-test-finished-with-test-result.md

```diff
@@ -0,0 +1,53 @@
+# Behavioral change of JMeter
+
+Behavioral change of the testing service in Keptn (jmeter-service) to not send evaluation-done events, but rather a test-finished events with a test result.
+
+## Motivation
+
+In the current implementation of the jmeter-service, the service sends a `sh.keptn.events.evaluation-done` event in case a test execution failed. In the data block of this event the property `result` is set to `fail`. However, sending this event type creates confusion because the jmeter-service did not execute an actual evaluation based on SLO or SLI configurations like the lighthouse-service does. Besides, the corresponding `sh.keptn.events.tests-finished` event is missing, and no status of the test execution is provided.
+
+## Explanation
+
+Instead of sending a `sh.keptn.events.evaluation-done` event for a failed test execution, this KEP proposes to send a `sh.keptn.events.tests-finished` event with the property `result`, which is set to `fail`. The following example shows the structure for this particular case:
+
+```json
+{
+ "data": {
+    "project": "sockshop",
+    "stage": "staging",
+    "service": "carts",
+    "testStrategy": "performance",
+    "deploymentStrategy": "direct",
+    "start": "2019-09-01 12:00:00",
+    "end": "2019-09-01 12:05:00",
+    "result": "fail"
+  },
+  "type": "sh.keptn.events.tests-finished",
+  "specversion": "0.2",
+  "source": "https://github.com/keptn/keptn/jmeter-service",
+  "id": "f2b878d3-03c0-4e8f-bc3f-454bc1b3d79d",
+  "time": "2019-06-07T07:02:15.64489Z",
+  "contenttype": "application/json",
+  "shkeptncontext": "08735340-6f9e-4b32-97ff-3b6c292bc509"
+}
+```
+
+## Internal details
+
+* *How the change would impact and interact with existing functionality?*: Since the jmeter-service does not send the `sh.keptn.events.evaluation-done`, the lighthouse-service is responsible for this step. To make sure, the lighthouse-service could receive the test result and skips the process of querying the SLI-provider since no monitoring data from the test would be available. 
```

Comment:
Actually, one could still want to query the SLI-provider for data, as monitoring data for the test could be available.

Reply:
Valid point! 
This is the reason why I used 'could' in the last sentence.

Reply:
Then I would like to ask you to move that sentence into the **Future possibilities** section (and to re-phrase it such that it can stand on its own).

Leave the part with 
> Since the jmeter-service does not send the `sh.keptn.events.evaluation-done`, the lighthouse-service is responsible for this step. 

as it is.

Reply:
Okay. I moved the sentence to the *Future possibilities* section.